File: src/main/kotlin/com/mino/smartcheck/config/ModelMapperConfig.java
package com.mino.smartcheck.config;

import com.mino.smartcheck.dto.UsuarioDto;
import com.mino.smartcheck.model.Usuario;
import org.modelmapper.ModelMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * @author Carlos Montoya
 * @since 14/10/2019
 */
@Configuration
public class ModelMapperConfig
{
	@Bean
	public ModelMapper modelMapper() {
		ModelMapper mapper = new ModelMapper();
		mapper.createTypeMap(Usuario.class, UsuarioDto.class)
				.addMappings(mapping -> mapping.skip(UsuarioDto::setPassword)); // password must be open
		return mapper;
	}
}

File: src/main/kotlin/com/mino/smartcheck/config/ModelMapperConfig.kt
package com.mino.smartcheck.config

import com.mino.smartcheck.dto.UsuarioDto
import com.mino.smartcheck.model.Usuario
import org.modelmapper.ModelMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ModelMapperConfig
{
	@Bean
	fun modelMapper(): ModelMapper {
		val mapper = ModelMapper()
		mapper
				.createTypeMap(UsuarioDto::class.java, Usuario::class.java)
				.addMapping(UsuarioDto::username, Usuario::setCorreo)
				.addMapping(UsuarioDto::password, Usuario::setContrasena)
		return mapper;
	}
}
